client/blocks/comment-detail/comment-detail-reply.jsx

```diff
@@ -13,15 +13,42 @@ import { get } from 'lodash';
 import AutoDirection from 'components/auto-direction';
 import { getCurrentUser } from 'state/current-user/selectors';
 
+const TEXTAREA_MAX_HEIGHT = 236; // 10 lines
+const TEXTAREA_MIN_HEIGHT_COLLAPSED = 47;
+const TEXTAREA_MIN_HEIGHT_FOCUSED = 68; // 2 lines
+const TEXTAREA_VERTICAL_BORDER = 2;
+
 export class CommentDetailReply extends Component {
 	state = {
 		commentText: '',
 		hasFocus: false,
+		textareaMinHeight: TEXTAREA_MIN_HEIGHT_COLLAPSED,
 	};
 
-	handleTextChange = event => this.setState( { commentText: event.target.value } );
+	bindTextareaRef = textarea => {
+		this.textarea = textarea;
+	}
+
+	calculateTextareaMinHeight = () => {
+		const textareaScrollHeight = this.textarea.scrollHeight;
+		const textareaMinHeight = Math.min( TEXTAREA_MAX_HEIGHT, textareaScrollHeight + TEXTAREA_VERTICAL_BORDER );
```

Comment:
Awesome suggestion - even though maybe it's not what you meant. 😄 
By replacing `min-height` with `height` I basically "unlocked" the resize to go lower than the current textarea height set via JS/CSS.

FYI: I can't set a fixed `max-height` because it blocks the manual resize.
The logic there is that the textarea can't be taller than `TEXTAREA_MAX_HEIGHT` _when resizing automatically_.
Also, that constant is needed to toggle the overflow/scrollbar (something that doesn't happen reliably in CSS alone).